sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -103,7 +103,15 @@ function appendPath(url: string, pathToAppend?: string): string {
     pathToAppend = pathToAppend.substring(1);
   }
 
-  newPath = newPath + pathToAppend;
+  const searchStart = pathToAppend.indexOf("?");
+  if (searchStart !== -1) {
+    const path = pathToAppend.substr(0, searchStart);
```

Comment:
nit: MDN prefers [substring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring), possibly because folks assume the second param is an index rather than length?